### DISCUSSION TITLE
Which is better PostgreSQL or CouchDB for future SW360?

### DISCUSSION THREAD

**Discussion by KoukiHama** [eclipse-sw360/sw360#2554] [General]
Which is better PostgreSQL or CouchDB for future SW360?

**Comment by arunazhakesan** [comment]
From Siemens POV, we will do a deep level analysis of the pros and cons and come back with the results here.

**Comment by KoukiHama** [comment]
(from sw360 telco)
![image](https://github.com/user-attachments/assets/aa782485-418d-4518-9bd2-3a5b9060e50d)

**Comment by losmateusz1** [comment]
Hello,

to revive this conversation, I prepared small list of arguments for migration to postgresql

1. Data Integrity & Reliability
Structured Data Control: Legal and compliance teams require structured data to ensure accuracy and completeness in reviews. PostgreSQL’s schema enforcement provides more control over data structure and integrity.
Referential Integrity: Ensures consistency between components, libraries, licenses, and vulnerabilities, preventing orphaned records, duplicate dependencies, and inconsistent metadata.
Point-in-Time Recovery (PITR): PostgreSQL allows for precise recovery of data in case of accidental deletion or corruption, ensuring business continuity.
2. Efficient Data Querying & Searching
Foreign Keys & Constraints: Ensures data validity and prevents broken references between resources.
Full-Text Search & Vector Search: PostgreSQL’s built-in full-text search and ability to integrate vector search(e.g., via pgvector) can enhance search functionality across large amounts of component, project, and license data.
3. Improved Reporting & Analytics
Optimized Complex Queries: PostgreSQL allows for recursive queries to efficiently retrieve dependency trees, reducing the need for additional processing.
Aggregations & Joins: Unlike NoSQL, where complex queries require extensive application-level computation, PostgreSQL can handle them natively using joins and aggregations.
